lib/sass/stack.rb

```diff
@@ -98,6 +98,16 @@ def with_mixin(filename, line, name)
       with_frame(filename, line, :mixin, name) {yield}
     end
 
+    # Pushes a function frame onto the stack.
+    #
+    # @param filename [String] See \{Frame#filename}.
+    # @param line [String] See \{Frame#line}.
+    # @param name [String] See \{Frame#name}.
+    # @yield [] A block in which the new frame is on the stack.
+    def with_function(filename, line, name)
+      with_frame(filename, line, :function, name) {yield}
+    end
```

Comment:
I think adding functions to the stack is the right thing to do—as far as I can tell the only reason they weren't before was that we didn't really need them. It doesn't look like it'll affect a huge amount of stuff, but it *will* affect the output of `@warn` when it's used within a function, and we should add specs for that.